test/integration/integration_test.h

```diff
@@ -4,24 +4,25 @@
 
 #include "gtest/gtest.h"
 
-class IntegrationTest : public BaseIntegrationTest, public testing::Test {
+class IntegrationTest : public BaseIntegrationTest,
+                        public testing::TestWithParam<Network::Address::IpVersion> {
 public:
   /**
-   * Global initializer for all integration tests.
+   * Initializer for individual integration tests.
    */
-  static void SetUpTestCase() {
+  void SetUp() override {
```

Comment:
Hmm, yeah. Forgot about this. Is the only reason we are doing static removal to support TEST_P? Is there any way to do TEST_P with different static setup/teardown? Like somehow instantiating twice? I guess I'm just wondering if the all the static removal and longer test time is really worth it (since we can already run the different binaries in parallel).